In [1]:
import requests
import os
import pandas as pd

# Foursquare

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [8]:
api_key = os.getenv('FOURSQUARE_API_KEY')

# Coordinates for Paris, France
latitude = 48.879359419425
longitude = 2.3665961623192

# Category ID for bike stations = '13000'
url = f"https://api.foursquare.com/v3/places/search?ll={latitude},{longitude}&radius=1000&categories=13000"  

headers = {
    'Authorization': api_key,
    'Accept': 'application/json'  
}

response = requests.get(url, headers=headers)

if response.status_code == 200:
    print("Success")
    places_data = response.json()  
    
    if 'results' in places_data:
        for place in places_data['results']:
            name = place.get('name', 'No name')
            location = place.get('location', {})
            address = location.get('address', 'No address provided')
            print(f"Place: {name}, Address: {address}")
    else:
        print("No bike stations found within the specified radius.")
else:
    print("Error message:", response.text)


Success
Place: Kheak & Véro, Address: 1 rue Alexandre Parodi
Place: El Nopal, Address: 3 rue Eugène Varlin
Place: Point Éphémère, Address: 200 quai de Valmy, 10e
Place: Bizz'Art, Address: 167 quai de Valmy
Place: Khayma rooftop, Address: 9-11 place du Colonel Fabien
Place: Les Enfants Perdus, Address: 9 rue des Récollets
Place: Le Comptoir Général, Address: 80-82 quai de Jemmapes
Place: Ten Belles, Address: 10 rue de la Grange aux Belles, 10e
Place: Early June, Address: 19 rue Jean Poulmarch
Place: Résidence Kann Coffee Shop, Address: 28 rue des Vinaigriers


Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [12]:
places_url = f"https://api.foursquare.com/v3/places/search?ll={latitude},{longitude}&radius=1000"

places_response = requests.get(places_url, headers=headers)

if places_response.status_code == 200:
    print("Success")
    data = places_response.json()

    if 'results' in data:
        for place in data['results']:
            name = place.get('name', 'No name')
            location = place.get('location', {})
            address = location.get('address', 'No address provided')
            city = location.get('city', 'No city provided')
            country = location.get('country', 'No country provided')
            rating = place.get('rating', 'No rating available')  

            print(f"Name: {name}")
            print(f"Location: {address}, {city}, {country}")
            print(f"Rating: {rating}")
            print('\n')

    else:
        print("No places found within the specified radius.")
else:
    print(f"Error message: {places_response.text}")

Success
Name: Kheak & Véro
Location: 1 rue Alexandre Parodi, No city provided, FR
Rating: No rating available


Name: Labranda
Location: 18 rue Louis Blanc, No city provided, FR
Rating: No rating available


Name: Street Bangkok Local Food
Location: 3 rue Eugene Varlin, No city provided, FR
Rating: No rating available


Name: El Nopal
Location: 3 rue Eugène Varlin, No city provided, FR
Rating: No rating available


Name: Khayma rooftop
Location: 9-11 place du Colonel Fabien, No city provided, FR
Rating: No rating available


Name: Point Éphémère
Location: 200 quai de Valmy, 10e, No city provided, FR
Rating: No rating available


Name: Canal Saint-Martin
Location: 100 quai de Valmy, No city provided, FR
Rating: No rating available


Name: Gumbo Yaya
Location: 3 rue Charles Robin, No city provided, FR
Rating: No rating available


Name: Panda Panda
Location: 21 rue Juliette Dodu, No city provided, FR
Rating: No rating available


Name: Bodhi Vegan
Location: 23 avenue Claude Vellefaux, No

Put your parsed results into a DataFrame

In [13]:
place_list = []

if 'results' in data:
        for place in data['results']:
            name = place.get('name', 'No name')
            location = place.get('location', {})
            address = location.get('address', 'No address provided')
            city = location.get('city', 'No city provided')
            country = location.get('country', 'No country provided')
            rating = place.get('rating', 'No rating available')  
            
            place_list.append({
                'Name': name,
                'Address': address,
                'City': city,
                'Country': country,
                'Rating': rating,
            })

df = pd.DataFrame(place_list)
print("\nPlaces DataFrame:")
print(df)


Places DataFrame:
                        Name                       Address              City  \
0               Kheak & Véro        1 rue Alexandre Parodi  No city provided   
1                   Labranda            18 rue Louis Blanc  No city provided   
2  Street Bangkok Local Food           3 rue Eugene Varlin  No city provided   
3                   El Nopal           3 rue Eugène Varlin  No city provided   
4             Khayma rooftop  9-11 place du Colonel Fabien  No city provided   
5             Point Éphémère        200 quai de Valmy, 10e  No city provided   
6         Canal Saint-Martin             100 quai de Valmy  No city provided   
7                 Gumbo Yaya           3 rue Charles Robin  No city provided   
8                Panda Panda          21 rue Juliette Dodu  No city provided   
9                Bodhi Vegan    23 avenue Claude Vellefaux  No city provided   

  Country               Rating  
0      FR  No rating available  
1      FR  No rating available  
2

# Yelp

Send a request to Yelp with a small radius (1000m) for all the bike stations in your city of choice. 

In [23]:
yelp_api_key = os.getenv('YELP_API_KEY')


latitude = 48.879359419425
longitude = 2.3665961623192

yelp_url = f"https://api.yelp.com/v3/businesses/search?latitude={latitude}&longitude={longitude}&radius=1000&categories=13000"  

headers = {
    'Authorization': f'Bearer {yelp_api_key}',
    'Accept': 'application/json'
}


places_response = requests.get(yelp_url, headers=headers)

if places_response.status_code == 200:
    print("Success")
    data = places_response.json()

    if 'businesses' in data and data['businesses']:
        for place in data['businesses']:
            name = place.get('name', 'No name')
            location = place.get('location', {})
            address = location.get('address', 'No address provided')
            print(f"Place: {name}, Address: {address}")
    else:
        print("No bike stations found within the specified radius.")
else:
    print("Error message:", response.text)

Success
Place: Restaurant le Robinet d'Or, Address: No address provided
Place: Street Bangkok Local Food, Address: No address provided
Place: FLESH, Address: No address provided
Place: Le Valmy, Address: No address provided
Place: Kheak & Vero, Address: No address provided
Place: Point Ephémère, Address: No address provided
Place: La Bulle, Address: No address provided
Place: John Viande, Address: No address provided
Place: Les Enfants Perdus, Address: No address provided
Place: Paname Brewing Company, Address: No address provided
Place: Assanabel, Address: No address provided
Place: Latinogourmand, Address: No address provided
Place: Le Chansonnier, Address: No address provided
Place: Bizz' Art, Address: No address provided
Place: Generator Paris, Address: No address provided
Place: Le Cafe Bizz'art, Address: No address provided
Place: Le 52, Address: No address provided
Place: Holybelly 5, Address: No address provided
Place: Le Réfectoire - Foodtruck, Address: No address provided
Pla

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [20]:
yelp_url = f"https://api.yelp.com/v3/businesses/search?latitude={latitude}&longitude={longitude}&radius=1000"

headers = {
    'Authorization': f'Bearer {yelp_api_key}',
    'Accept': 'application/json'
}

places_response = requests.get(yelp_url, headers=headers)

if places_response.status_code == 200:
    print("Success")
    data = places_response.json()

    if 'businesses' in data and data['businesses']:
        for place in data['businesses']:
            name = place.get('name', 'No name')
            rating = place.get('rating', 'No rating available')
            location = place.get('location', {})
            address = ", ".join([str(location.get('address1', 'No address provided')),
                                 str(location.get('address2', '')),
                                 str(location.get('address3', ''))]).strip(', ')
            city = location.get('city', 'No city provided')
            country = location.get('country', 'No country provided')

            print(f"Name: {name}")
            print(f"Location: {address}, {city}, {country}")
            print(f"Rating: {rating}")
            print(f"\n")
    else:
        print("No places found within the specified radius.")
else:
    print(f"Error message: {places_response.text}")
    print(f"Status Code: {places_response.status_code}")

Success
Name: Restaurant le Robinet d'Or
Location: 7 rue Eugène Varlin, Paris, FR
Rating: 4.1


Name: Street Bangkok Local Food
Location: 3 rue Eugène Varlin, Paris, FR
Rating: 4.1


Name: FLESH
Location: 23 rue Louis Blanc, , None, Paris, FR
Rating: 4.6


Name: Le Valmy
Location: 145 quai de Valmy, None, None, Paris, FR
Rating: 3.9


Name: Kheak & Vero
Location: 1 rue Alexandre Parodi, Paris, FR
Rating: 4.3


Name: Point Ephémère
Location: 200 quai de Valmy, Paris, FR
Rating: 3.8


Name: La Bulle
Location: 48 rue Louis Blanc, None, None, Paris, FR
Rating: 4.0


Name: John Viande
Location: 7bis rue de Paradis, None, Paris, FR
Rating: 4.5


Name: Les Enfants Perdus
Location: 9 rue des Recollets, Paris, FR
Rating: 4.1


Name: Paname Brewing Company
Location: 41 bis quai de la Loire, Paris, FR
Rating: 4.2


Name: Assanabel
Location: 6 rue Pierre Chausson, Paris, FR
Rating: 4.4


Name: Latinogourmand
Location: 233 rue la Fayette, Paris, FR
Rating: 4.5


Name: Le Chansonnier
Location: 14 Ru

Put your parsed results into a DataFrame

In [25]:
place_list = []

if 'businesses' in data and data['businesses']:
        for place in data['businesses']:
            name = place.get('name', 'No name')
            rating = place.get('rating', 'No rating available')
            location = place.get('location', {})
            address = ", ".join([str(location.get('address1', 'No address provided')),
                                 str(location.get('address2', '')),
                                 str(location.get('address3', ''))]).strip(', ')
            city = location.get('city', 'No city provided')
            country = location.get('country', 'No country provided')
            
            place_list.append({
                'Name': name,
                'Address': address,
                'City': city,
                'Country': country,
                'Rating': rating,
            })

df = pd.DataFrame(place_list)
print("\nPlaces DataFrame:")
print(df)


Places DataFrame:
                          Name                             Address   City  \
0   Restaurant le Robinet d'Or                 7 rue Eugène Varlin  Paris   
1    Street Bangkok Local Food                 3 rue Eugène Varlin  Paris   
2                        FLESH          23 rue Louis Blanc, , None  Paris   
3                     Le Valmy       145 quai de Valmy, None, None  Paris   
4                 Kheak & Vero              1 rue Alexandre Parodi  Paris   
5               Point Ephémère                   200 quai de Valmy  Paris   
6                     La Bulle      48 rue Louis Blanc, None, None  Paris   
7                  John Viande           7bis rue de Paradis, None  Paris   
8           Les Enfants Perdus                 9 rue des Recollets  Paris   
9       Paname Brewing Company             41 bis quai de la Loire  Paris   
10                   Assanabel               6 rue Pierre Chausson  Paris   
11              Latinogourmand                  233 rue l

# Comparing Results

Which API provided you with more complete data? Provide an explanation. 

Get the top 10 restaurants according to their rating


Top 10 Restaurants by Rating:
                        Name                         Address   City Country  \
0                Holybelly 5    5 rue Lucien Sampaix, , None  Paris      FR   
1                      FLESH      23 rue Louis Blanc, , None  Paris      FR   
2                John Viande       7bis rue de Paradis, None  Paris      FR   
3             Latinogourmand              233 rue la Fayette  Paris      FR   
4                      Le 52  52 rue du Faubourg Saint-Denis  Paris      FR   
5  Le Réfectoire - Foodtruck                                  Paris      FR   
6                  Assanabel           6 rue Pierre Chausson  Paris      FR   
7               Kheak & Vero          1 rue Alexandre Parodi  Paris      FR   
8     Paname Brewing Company         41 bis quai de la Loire  Paris      FR   
9                     Siseng             82 quai de Jemmapes  Paris      FR   

   Rating  
0     4.7  
1     4.6  
2     4.5  
3     4.5  
4     4.5  
5     4.5  
6     4.4  
7     4.3  
8     4.2  
9     4.2  

Yelp API provides more data as it contains the raiting for the restaurants. 